### ISSUE TITLE
Update list view of search results

### ISSUE THREAD

**Issue by ggeisler** [projectblacklight/arclight#669]
Part of #667. 

Note that we intend for the arrangement of search result item elements to very closely match the arrangement of elements in the Collection contents/context tab (the tree view of the collection; see #670) and the list of items in the Online items tab (tickets with designs for those forthcoming).

There are a bunch of icons shown in the mockups below. We still need to decide on final icons and define sizes, colors, etc. (#16). I think the recommendations in this ticket can be completed without worrying about icons (with the caveat that some CSS adjustments might be necessary after we add the icons in later), or you could use some placeholder icons.

## Desktop layout of search results

<img width="877" alt="Screen Shot 2019-08-28 at 4 36 48 PM" src="https://user-images.githubusercontent.com/101482/63900031-11b6ce00-c9b4-11e9-8809-6c47cee4f009.png">

---

**A: Result item label** - No change from current, except:

- [x] Prefix the label with an icon conveying whether the item is a collection, a file (lowest-level component), or something in between those two levels.

**B: Result item breadcrumb** - This is similar to the breadcrumb shown in the current search results display. What's changed: 

- [x] Add the repository to the beginning of the breadcrumb
- [x] Move the breadcrumb so it is the last row of the result item entry
- [x] Change the link styling to remove the link color. The breadcrumb elements should still be individually linked as they are now, we're just trying to reduce the amount of blue (future styling updates TBD).

**C: Online content label** - This is new and should match what we do in the collection contents and online contents tabs. Not all items will have this label.

- [x] If the result item has digital content and the item is a file, show a green image icon.
- [x] If the result item is a parent container that either has digital content or contains at least one child with digital content, show the text "Online content" in green (#28A745).

**D: Bookmark checkbox** - We currently show a bookmark checkbox for each result item. All items should continue to have a bookmark checkbox. What's changed:

- [x] Move the bookmark checkbox so it is right-aligned at the edge of the content area, on the same line as the result item label.
- [x] Replace the text "Bookmark" with a ribbon icon.

**E: Container label** - If the result item has a container label:

- [x] Display the label value on the same line as the result item title
- [x] Use a muted text color for the container label: #ADB5BD

See the third result item in the mockup above for the desired order of the container label, online content, and bookmark checkbox elements.

**F: Extent label** - We currently show the value of "extent" as part of the result item metadata. What's changed:

- [x] Move the extent value, if it exists for the result item, to come directly after the result item label, on the same line
- [x] Style the extent label as a Bootstrap badge, using the styling we currently use for the result item header section (see CSS class `al-document-title-bar`)

**G: Scope note/abstract** - No change from what we currently do, except:

- [x] Add a max-width of 45em to the abstract/scope note for readability

**Hit highlighting** - If there is a snippet to show, no change from what we currently do; it just goes directly beneath the result item label:

<img width="877" alt="Screen Shot 2019-08-28 at 5 27 55 PM" src="https://user-images.githubusercontent.com/101482/63901429-61e45f00-c9b9-11e9-855b-92997c95f41b.png">

## Adjustments for mobile 

- [x] Reduce size of result item "level" icon (the icon to the left of the result item title/label). Details TBD.

The only other significant adjustment we need to worry about in terms of responsive sizing has to do with the search result item labels and bookmark checkbox. The goal here is exactly the same as described for the collection context tree described in #670.

- [x] At the `small` breakpoint and below, the labels and bookmark checkbox should move to the bottom row of the result item entry:

<img width="799" alt="Screen Shot 2019-08-29 at 8 51 43 AM" src="https://user-images.githubusercontent.com/101482/63956528-723d1e00-ca3b-11e9-807b-64e39eefcfe6.png">

- [x] The online label and bookmark checkbox should be right-aligned in the row
- [x] If there is a container label (e.g., "Box 3, Folder 99"), it should be left-aligned in the row

<img width="392" alt="Screen Shot 2019-08-29 at 8 48 26 AM" src="https://user-images.githubusercontent.com/101482/63956626-a87a9d80-ca3b-11e9-990a-73041eafa2a5.png">

If a hit highlighted snippet is shown, its placement remains directly beneath the result item label.

**Comment by labradford** [comment]
Can we get svg icon files for the new icons? Or do you want to use a library?
Going to put this ticket back to ready so I don't get blocked on it.

**Comment by mejackreed** [comment]
We will use svg icons, but its ok to wait on that piece in a future ticket.

**Comment by labradford** [comment]
@mejackreed With this ticket being so big, should we break it into smaller pieces?

**Comment by mejackreed** [comment]
All of the items in this ticket seem to have been completed. Let's track issues/ updates in a new issue and link back if needed.
